src/app/clusters/level-control/level-control.cpp

```diff
@@ -123,6 +123,137 @@ static bool shouldExecuteIfOff(EndpointId endpoint, CommandId commandId,
                                chip::Optional<chip::BitMask<LevelControlOptions>> optionsMask,
                                chip::Optional<chip::BitMask<LevelControlOptions>> optionsOverride);
 
+#ifdef EMBER_AF_PLUGIN_SCENES
+class DefaultLevelControlSceneHandler : public scenes::DefaultSceneHandlerImpl
+{
+public:
+    // As per spec, 2 attributes are scenable in the level control cluster
+    static constexpr uint8_t kLevelMaxScenableAttributes = 2;
+
+    DefaultLevelControlSceneHandler() = default;
+    ~DefaultLevelControlSceneHandler() override {}
+
+    // Default function for LevelControl cluster, only puts the LevelControl cluster ID in the span if supported on the caller
+    // endpoint
+    virtual void GetSupportedClusters(EndpointId endpoint, Span<ClusterId> & clusterBuffer) override
+    {
+        if (emberAfContainsServer(endpoint, LevelControl::Id) && clusterBuffer.size() >= 1)
+        {
+            clusterBuffer[0] = LevelControl::Id;
+            clusterBuffer.reduce_size(1);
+        }
+        else
+        {
+            clusterBuffer.reduce_size(0);
+        }
+    }
+
+    // Default function for LevelControl cluster, only checks if LevelControl is enabled on the endpoint
+    bool SupportsCluster(EndpointId endpoint, ClusterId cluster) override
+    {
+        return (cluster == LevelControl::Id) && (emberAfContainsServer(endpoint, LevelControl::Id));
+    }
+
+    /// @brief Serialize the Cluster's EFS value
+    /// @param endpoint target endpoint
+    /// @param cluster  target cluster
+    /// @param serializedBytes data to serialize into EFS
+    /// @return CHIP_NO_ERROR if successfully serialized the data, CHIP_ERROR_INVALID_ARGUMENT otherwise
+    CHIP_ERROR SerializeSave(EndpointId endpoint, ClusterId cluster, MutableByteSpan & serializedBytes) override
+    {
+        using AttributeValuePair = Scenes::Structs::AttributeValuePair::Type;
+
+        app::DataModel::Nullable<uint8_t> level;
+        VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == Attributes::CurrentLevel::Get(endpoint, level), CHIP_ERROR_READ_FAILED);
+
+        AttributeValuePair pairs[kLevelMaxScenableAttributes];
+
+        uint8_t maxLevel;
+        VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == Attributes::MaxLevel::Get(endpoint, &maxLevel), CHIP_ERROR_READ_FAILED);
+
+        pairs[0].attributeID    = Attributes::CurrentLevel::Id;
+        pairs[0].attributeValue = (!level.IsNull()) ? level.Value() : maxLevel + 1;
```

Comment:
This will be broken if the scene is then recalled with a different maxlevel for some reason.

Is there a reason this is not using the existing attribute-storage-null-handling.h facilities which would let us encode null in an integer?